repos/system_upgrade/el7toel8/libraries/mounting.py

```diff
@@ -140,11 +141,14 @@ def full_path(self, path):
         """
         Transform the path given to be prefixed with the base_dir, to get the real path on the system.
 
+        The function is secured, so it is not possible to return path outside
+        of the self.base_dir directory.
+
         Example: self.base_dir = '/var/lib/leapp/scratch/userspace'
                  path = '/etc/yum.repos.d/redhat.repo'
                  The result would be: /var/lib/leapp/scratch/userspace/etc/yum.repos.d/redhat.repo
         """
-        return os.path.join(self.base_dir, path.lstrip('/'))
+        return os.path.join(self.base_dir, os.path.abspath(path).lstrip('/'))
```

Comment:
Could you please elaborate more on this change?

Reply:
Probably you missed the commit message, otherwise I am not sure what more I could write about that:
```
    mounting library: secure the isolated actions
    
    Previously it was possible by mistake do actions out of the isolated
    environment, e.g.
        context.copy_to('/path/src', '../../path/dst')
    
    In such a case, the dst path is out of the basedir of the isolated
    environment. To prevent this, update the IsolatedActions.full_path
    method.
    
    Currently, the solution transfer the path "inside the env"
    to absolute path and then the join of paths is applied. As it is
    expected to use always absolute paths, it should not be problem
    in general.
    
    Warning:
      But in case someone uses a relative path, it generates
      completely different path than expected. If needed, it
      can be replaced by solution where relative paths are unchanged,
      but in such a case, the method has to raise an exception when
      the path is in real out of the environment.

```

Reply:
I want to keep this fixed as in the PR as with the provided functionality we need to be much sure that user/we/someone.. will not do something wrong what could negatively affect the host system.